src/components/UserProfile/UserProfile.tsx

```
@@ -0,0 +1,522 @@
+          Please log in to view profile information.
+        </p>
+        <a href="/login" className="btn btn-primary mt-3">
+          <i className="fas fa-sign-in-alt me-2" />
+          Go to Login
+        </a>
+      </div>
+    );
+  }
+
+  // Determine which query to use based on whether it's own profile or not
+  const { data, loading, error, refetch } = useQuery(
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)